src/ResourceManager/StorageSync/Commands.StorageSync/PowerShellCommandRunner.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace Microsoft.Azure.Commands.StorageSync.Evaluation
+{
+    using Interfaces;
+    using System.Collections.ObjectModel;
+    using System.Management.Automation;
+    using System.Management.Automation.Runspaces;
+
+    public class PowerShellCommandRunner : IPowershellCommandRunner
+    {
+        #region Fields and Properties
+        private readonly Runspace _runspace;
+        private readonly PowerShell _powerShell;
+        #endregion
+
+        #region Constructors
+        public PowerShellCommandRunner(string computerName, PSCredential credential)
+        {
+            WSManConnectionInfo connectionInfo = new WSManConnectionInfo();
+
+            if (computerName != null && !computerName.Equals("localhost", System.StringComparison.OrdinalIgnoreCase))
+            {
+                connectionInfo.ComputerName = computerName;
+            }
+
+            if (credential != null)
+            {
+                connectionInfo.Credential = credential;
+            }
+
+            this._runspace = RunspaceFactory.CreateRunspace(connectionInfo);
+            this._runspace.Open();
+
+            this._powerShell = PowerShell.Create();
```

Comment:
@afedyashov why do we need to create a PowerShell instance? This is something we should avoid doing

Reply:
we are using PS Remoting to execute validations of remote machines, currently we execute file system checks (we require NTFS) as well as OS version and edition checks. In the future, we'll be adding backup and antivirus software checks, etc. All such checks are implemented as scripts running over remote machine over a single PowerShell/RunSpace that we've setup once per execution. If customers chose not to perform environment evaluation, this code is not running.